modules/core/src/main/java/org/locationtech/jts/io/WKBWriter.java

```diff
@@ -287,10 +303,57 @@ public WKBWriter(int outputDimension, int byteOrder, boolean includeSRID) {
       this.byteOrder = byteOrder;
       this.includeSRID = includeSRID;
       
-      if (outputDimension < 2 || outputDimension > 3)
-        throw new IllegalArgumentException("Output dimension must be 2 or 3");
+      if (outputDimension < 2 || outputDimension > 4)
+        throw new IllegalArgumentException("Output dimension must be 2 to 4");
+
+      this.outputOrdinates = EnumSet.of(Ordinate.X, Ordinate.Y);
+      if (outputDimension > 2)
+        outputOrdinates.add(Ordinate.Z);
+      if (outputDimension > 3)
+        outputOrdinates.add(Ordinate.M);
   }
-  
+
+  /**
+   * Sets the {@link Ordinate} that are to be written. Possible members are:
+   * <ul>
+   * <li>{@link Ordinate#X}</li>
+   * <li>{@link Ordinate#Y}</li>
+   * <li>{@link Ordinate#Z}</li>
+   * <li>{@link Ordinate#M}</li>
+   * </ul>
+   * Values of {@link Ordinate#X} and {@link Ordinate#Y} are always assumed and not
+   * particularly checked for.
+   *
+   * @param outputOrdinates A set of {@link Ordinate} values
+   */
+  public void setOutputOrdinates(EnumSet<Ordinate> outputOrdinates) {
```

Comment:
Minor feedback that having the same parameter name as field name makes the method harder to read through.

It is interesting that a lot of effort is done to update the existing `this.outputOrdinates` field in place, rather than just use the provided `outputOrdinates` parameter.